File: frontend/lib/features/feedback/domain/entities/feedback_post.dart
import 'package:equatable/equatable.dart';
import 'package:budget_book/features/feedback/domain/entities/feedback_comment.dart';

class FeedbackPost extends Equatable {
  final String id;
  final String userId;
  final String userName;
  final String category;
  final String title;
  final String content;
  final String status;
  final String? adminNote;
  final String? resolvedReleaseId;
  final List<FeedbackComment> comments;
  final DateTime createdAt;
  final DateTime updatedAt;

  const FeedbackPost({
    required this.id,
    required this.userId,
    required this.userName,
    required this.category,
    required this.title,
    required this.content,
    required this.status,
    this.adminNote,
    this.resolvedReleaseId,
    required this.comments,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        userId,
        userName,
        category,
        title,
        content,
        status,
        adminNote,
        resolvedReleaseId,
        comments,
        createdAt,
        updatedAt,
      ];
}

class FeedbackStats extends Equatable {
  final int totalCount;
  final Map<String, int> byCategory;
  final Map<String, int> byStatus;

  const FeedbackStats({
    required this.totalCount,
    required this.byCategory,
    required this.byStatus,
  });

  @override
  List<Object?> get props => [totalCount, byCategory, byStatus];
}

File: backend/src/main/kotlin/com/budgetbook/feedback/domain/FeedbackPost.kt
package com.budgetbook.feedback.domain

import com.budgetbook.auth.domain.User
import com.budgetbook.common.entity.BaseTimeEntity
import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.FetchType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.OrderBy
import jakarta.persistence.Table
import java.util.UUID

@Entity
@Table(name = "feedback_posts")
class FeedbackPost(
    @Id
    val id: UUID = UUID.randomUUID(),

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    val user: User,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    var category: FeedbackCategory,

    @Column(nullable = false, length = 200)
    var title: String,

    @Column(nullable = false, columnDefinition = "TEXT")
    var content: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    var status: FeedbackStatus = FeedbackStatus.SUBMITTED,

    @Column(name = "admin_note", columnDefinition = "TEXT")
    var adminNote: String? = null,

    @Column(name = "resolved_release_id")
    var resolvedReleaseId: UUID? = null,

    @OneToMany(mappedBy = "post", cascade = [CascadeType.ALL], orphanRemoval = true)
    @OrderBy("createdAt ASC")
    val comments: MutableList<FeedbackComment> = mutableListOf()
) : BaseTimeEntity()
